types/mojang-net/index.d.ts

```diff
@@ -0,0 +1,172 @@
+     * are commonly used to create a new resource that is a
+     * subordinate of the specified URI.
+     */
+    GET = 2,
+    /**
+     * Represents the method for an HTTP HEAD request. HEAD
+     * requests are similar to a GET request, but are commonly used
+     * to retrieve just the HTTP response headers from the
+     * specified URI, and not the body contents.
+     */
+    DELETE = 3,
+    /**
+     * Represents the method for an HTTP PUT request. GET requests
+     * are commonly used to retrieve information about a resource
+     * at the specified URI.
+     */
+    HEAD = 4,
+}
+export class HttpClient {
+    /**
+     * @remarks
+     * Cancels all pending requests.
+     * @param reason
+     */
+    cancelAll(reason: string): void;
+    /**
+     * @remarks
+     * Performs a simple HTTP get request.
+     * @param uri
+     * URL to make an HTTP Request to.
+     * @returns
+     * An awaitable promise that contains the HTTP response.
+     */
+    get(uri: string): Promise<HttpResponse>;
+    /**
+     * @remarks
+     * Performs an HTTP request.
+     * @param config
+     * Contains an HTTP Request object with configuration data on
+     * the HTTP request.
+     * @returns
+     * An awaitable promise that contains the HTTP response.
+     */
+    request(config: HttpRequest): Promise<HttpResponse>;
+    protected constructor();
+}
+/**
+ * Represents an HTTP header - a key/value pair of
+ * meta-information about a request.
+ */
+export class HttpHeader {
+    /**
+     * Key of the HTTP header.
+     */
+    'key': string;
+    /**
+     * Value of the HTTP header.
+     */
+    'value': string;
+    constructor(key: string, value: mojangminecraftserveradmin.SecretString | string);
+}
+/**
+ * Main object for structuring a request.
+ */
+export class HttpRequest {
+    /**
+     * Content of the body of the HTTP request.
+     */
+    'body': string;
+    /**
+     * A collection of HTTP headers to add to the outbound request.
+     */
+    'headers': HttpHeader[];
+    /**
+     * HTTP method (e.g., GET or PUT or PATCH) to use for making
+     * the request.
+     */
+    'method': HttpRequestMethod;
+    /**
+     * Amount of time, in milliseconds, before the request times
+     * out and is abandoned.
+     */
+    'timeout': number;
+    /**
+     * The HTTP resource to access.
+     */
+    'uri': string;
+    /**
+     * @remarks
+     * Adds an additional header to the overall list of headers
+     * used in the corresponding HTTP request.
+     * @param key
+     * @param value
+     */
+    addHeader(key: string, value: string): HttpRequest;
+    constructor(uri: string);
+    setBody(body: string): HttpRequest;
+    /**
+     * @remarks
+     * Replaces and applies a set of HTTP Headers for the request.
+     * @param headers
+     */
+    setHeaders(headers: HttpHeader[]): HttpRequest;
+    setMethod(method: HttpRequestMethod): HttpRequest;
+    setTimeout(timeout: number): HttpRequest;
```

Comment:
Should this be documented?

Reply:
We'll add documentation in subsequent PRs beyond this one.